Extensions/CodeGeneration/CodeGenerationDisplayDriver.cs

```diff
@@ -131,11 +131,53 @@ private string ConvertJToken(JToken jToken)
                         string => $"\"{value}\"",
                         _ => value?.ToString()?.Replace(',', '.'), // Replace decimal commas.
                     };
+
                 case JArray jArray:
-                    return $"new[] {{ {string.Join(", ", jArray.Select(ConvertJToken))} }}";
+                    var format = $"{string.Join(string.Empty, Enumerable.Repeat(" ", indentationDepth + 4))}";
+
+                    var items = jArray.Select(item => ConvertJToken(item, indentationDepth + 8)).ToList();
+
+                    // If the items are formatted (for ListValueOption) the don't inject line-by-line formatting
+                    if (items.Any(item => item.Contains(Environment.NewLine, StringComparison.OrdinalIgnoreCase)))
+                    {
+                        var token = string.Join(string.Empty, items);
+                        return $"new[]\n{format}{{\n{token}{format}}}";
+                    }
+
+                    // Otherwise, make sure that we have proper formatting for string arrays
+                    var stringArrayCodeBuilder = new StringBuilder("new[]");
+                    stringArrayCodeBuilder.AppendLine();
+                    stringArrayCodeBuilder.AppendLine($"{format}{{");
+
+                    var itemFormat = $"{string.Join(string.Empty, Enumerable.Repeat(" ", indentationDepth + 8))}";

```

Comment:
```suggestion
                    var itemIndentation = $"{string.Join(string.Empty, Enumerable.Repeat(" ", indentationDepth + 8))}";
```

Reply:
And this one? I've updated mine to reflect the variable renaming that @Piedone suggested.